src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -4107,10 +4107,32 @@ QStringList QgsOgrProvider::uniqueStringsMatching( int index, const QString &sub
   QgsCPLHTTPFetchOverrider oCPLHTTPFetcher( mAuthCfg );
   QgsSetCPLHTTPFetchOverriderInitiatorClass( oCPLHTTPFetcher, QStringLiteral( "QgsOgrProvider" ) );
 
+  // uniqueStringsMatching() is supposed to be case insensitive, so use the
+  // ILIKE operator when it is available.
+  // Prior to GDAL 3.1, with OGR SQL, LIKE behaved like ILIKE
+  bool supportsILIKE = false;
+  {
+    QByteArray sql = "SELECT 1 FROM ";
+    sql += quotedIdentifier( mOgrLayer->name() );
+    sql += " WHERE 'a' ILIKE 'A' LIMIT 1";
```

Comment:
I'm guessing you've gone with this approach over a version check is to handle the drivers which don't use OGR sql?

Reply:
> I'm guessing you've gone with this approach over a version check is to handle the drivers which don't use OGR sql?

exactly.